server/src/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -852,7 +852,8 @@ public boolean releaseSecondaryIpFromNic(long ipAddressId) {
                 throw new InvalidParameterValueException("Can' remove the ip " + secondaryIp + "is associate with static NAT rule public IP address id " + publicIpVO.getId());
             }
 
-            if (_lbService.isLbRuleMappedToVmGuestIp(secondaryIp)) {
+            List<Long> lbRuleIdList = _firewallDao.listIdByNetworkAndPurposeAndNotRevoked(network.getId(), Purpose.LoadBalancing);
```

Comment:
@niteshsarda sorry what I said is not clear. You added two methods in Daos: listIdByNetworkAndPurposeAndNotRevoked and listByLoadBalancerIdVmIdAndInstanceIp, I suggest to implement it by one method in LoadBalancerDaoImpl.java